api/middleware/contentLengthLimiter.go

```diff
@@ -0,0 +1,84 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/multiversx/mx-chain-go/api/shared"
+
+	"github.com/multiversx/mx-multi-factor-auth-go-service/config"
+)
+
+const (
+	unknownContentLengthSize = -1
+	minSizeBytes             = 10
+)
+
+type contentLengthLimiter struct {
+	maxContentLengths map[string]uint64
+}
+
+// NewContentLengthLimiter will abort all requests that have Content-Length size bigger
+// than the one specified in config.
+func NewContentLengthLimiter(apiPackages map[string]config.APIPackageConfig) (*contentLengthLimiter, error) {
+	maxContentLengths := map[string]uint64{}
+	for group, groupCfg := range apiPackages {
+		groupPath := fmt.Sprintf("/%s", group)
+
+		for _, r := range groupCfg.Routes {
+			if r.MaxContentLength < minSizeBytes {
+				return nil, fmt.Errorf("%w, min expected %d, received %d", ErrMaxContentLengthTooSmall, minSizeBytes,
+					r.MaxContentLength)
+			}
+			fullPath := fmt.Sprintf("%s%s", groupPath, r.Name)
+			maxContentLengths[fullPath] = r.MaxContentLength
+		}
+	}
+
+	return &contentLengthLimiter{
+		maxContentLengths: maxContentLengths,
+	}, nil
+
+}
+
+// MiddlewareHandlerFunc returns the handler func used by the gin server when processing requests.
+func (r *contentLengthLimiter) MiddlewareHandlerFunc() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		size := c.Request.ContentLength
```

Comment:
we need an early exit here for get requests, and ignore my previous comment with the min value added to config
```go
		if c.Request.Method == http.MethodGet {
			return
		}
```